decorator.rb

```diff
@@ -0,0 +1,14 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  attr_accessor :nameable
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
[OPTIONAL]
Hello @waelelsafty07,

Great job on your `Decorator` class! 👍 Everything seems to be in order. However, I'd like to offer a small suggestion for code organization.

It's generally a good practice to place `attr_accessor` and similar macros (like `attr_reader` or `attr_writer`) towards the top of your class definition, just after any `include` or `extend` statements, but before method definitions. This makes it easier for other developers to quickly understand the attributes and the public interface of your class at a glance.

Consider rearranging your class like this:

```ruby
require_relative 'nameable'

class Decorator < Nameable
  attr_accessor :nameable

  def initialize(nameable)
    super()
    @nameable = nameable
  end

  def correct_name
    @nameable.correct_name
  end
end
```

This change is optional but could enhance the readability and maintainability of your code.

Keep up the good work! 😊

Warm regards,
Temitope